API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,101 @@
+    [Route("api/vl")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookService _bookService;
+
+        public BooksController(IBookService bookService)
+        {
+            _bookService = bookService;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns>A list of all books</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> GetBooks()
+        {
+            var books = await _bookService.GetBooksAsync();
+            if (books.Count == 0) return NotFound();
+            return Ok(books);
+        }
+
+        /// <summary>
+        /// Get a specific book by id
+        /// </summary>
+        /// <param name="id">id</param>
+        /// <returns>Book</returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<ActionResult<Book>> GetBook(int id)
+        {
+            var book = await _bookService.GetBookAsync(id);
+            if (book == null) return NotFound();
+            return Ok(book);
+        }
+
+        /// <summary>
+        /// Edit a book
+        /// </summary>
+        /// <param name="book">book</param>
+        /// <returns>Updated book</returns>
+        [HttpPut("{id}"), Authorize("IsLibrarian")]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> UpdateBook(UpdateBookDto book)
+        {
+            var bookUpdate = await _bookService.UpdateBookAsync(book);
+            return Accepted(bookUpdate);
+        }
+
+        /// <summary>
+        /// Create a new book
+        /// </summary>
+        /// <param name="book">book</param>
+        /// <returns>Newly created book</returns>
+        [HttpPost, Authorize("IsLibrarian")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesDefaultResponseType]
+        public async Task<CreatedResult> CreateBook(UpdateBookDto book)
+        {
+            var newBook = await _bookService.CreateBookAsync(book);
+            return Created("New Book", newBook);
+        }
+
+        /// <summary>
+        /// Delete a book by id
+        /// </summary>
+        /// <param name="id">id</param>
+        /// <returns>"Book was successfully deleted"</returns>
+        [HttpDelete("{id}"), Authorize("IsLibrarian")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> DeleteBook(int id)
```

Comment:
What are you returning here in your `IActionResult`?

Reply:
Ok or Not found

Reply:
I meant to put this on `public async Task<CreatedResult> CreateBook(BookDto book)`

Reply:
Which if you sent a 200, we know the book was successfully deleted

You can just return Ok()

Reply:
With ternary, this just becomes

`return result ? Ok() : NotFound();`

Reply:
![image](https://user-images.githubusercontent.com/99216796/196813669-c912bcb5-e2fe-4342-9fd7-1864fb824d3a.png)

Reply:
Talking about the HttpDelete here

Reply:
Is there a change needed here? I am confused now.

Reply:
I am returning a boolean to tell us if the book was deleted or not

Reply:
![image](https://user-images.githubusercontent.com/99216796/198628730-8cc77854-13e8-4aac-967f-52671467c2c1.png)

Reply:
![image](https://user-images.githubusercontent.com/99216796/198628857-8c3a7ee2-0ade-46d5-a8ad-3fe10b473ad9.png)

Reply:
If a change was written to the db, the result will be > 0 and return true.

Reply:
![image](https://user-images.githubusercontent.com/99216796/198629633-69a0e934-c6c5-4ed3-aec8-d3e30cb4bd50.png)

Reply:
that bool return then dictates which piece of the ternary triggers